R/tribble.R

```diff
@@ -27,19 +27,75 @@
 #'   "b", 4:6
 #' )
 tribble <- function(...) {
+  data <- extract_frame_data_from_dots(...)
 
+  if (length(data$frame_rest) == 0) {
+    out <- rep(list(logical()), length(data$frame_names))
+    names(out) <- data$frame_names
+    return(as_tibble(out))
+  }
+
+  frame_mat <- matrix(data$frame_rest, ncol = length(data$frame_names), byrow = TRUE)
+
+  frame_col <- lapply(seq_len(ncol(frame_mat)), function(i) {
```

Comment:
I could, but would appreciate more pointers about why. The implementation after `extract_frame_data_from_dots()` is very unique, so if we extract everything after it into a function, this function won't be reused anywhere but here. Note that the implementation of `tribble` and `frame_matrix` are quite different after `extract_frame_data_from_dots(...)`, so I don't think any reuse is possible.